models/clm/shell_scripts/cesm2_2/CLM5_hybrid_freerun

```diff
@@ -468,21 +486,30 @@ else     # CONTINUE_RUN == FALSE
 
    @ inst=1
    while (\$inst <= $num_instances)
+     
+      if ( \${num_instances} == 1) then
+         set inst_string = \`printf _%04d \$inst\`   
+         set inst_string2 = ''
+      else
+         set inst_string = \`printf _%04d \$inst\`
+         set inst_string2 = \`printf _%04d \$inst\`
```

Comment:
Unless I am mis-reading this (always possible)`inst_string` and `inst_string2` are identical.

I would just have `inst_string` if they are identical.

Reply:
Yes -- inst_string and inst_string2 are identical here, but only for the multi-instance case (see 'if' statement in lines 490-96).   On the other hand, In a single instance case these strings are different, because in a single instance case there is no appending of the instance number within the restart file name for the first time step.  Agreed --  it's confusing.    Other option would be to give multiple 'LINK' commands in a if statement, but that would generate more code.

Reply:
my bad, looks good as is. I'd missed the single instance case.